src/Symfony/Component/Security/Http/Authenticator/InteractiveAuthenticatorInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+/**
+ * This is an extension of the authenticator interface that must
+ * be used by interactive authenticators.
+ *
+ * Interactive login requires explicit user action (e.g. a login
+ * form or HTTP basic authentication).
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+interface InteractiveAuthenticatorInterface extends AuthenticatorInterface
+{
+    /**
+     * Should return true to make this authenticator perform
+     * an interactive login.
+     */
+    public function isInteractive(): bool;
+
+    /**
+     * Called when authentication executed and was successful!
+     *
+     * This should return the Response sent back to the user, like a
+     * RedirectResponse to the last page they visited.
+     *
+     * If you return null, the current request will continue, and the user
+     * will be authenticated. This makes sense, for example, with an API.
+     */
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $providerKey): ?Response;
```

Comment:
Now that I've investigated the `INTERACTIVE_LOGIN` event more, I think that `onAuthenticationSuccess()` should be moved from here back into `AuthenticatorInterface`. Two reasons:

1) We may deprecate this class AND the INTERACTIVE event later. I'm not sure it makes sense - especially because were going to need to make `isInteractive()` return true/false in weird circumstances to match the old system.

2) I think it makes sense for `AuthenticatorInterface` to have both `onAuthenticationSuccess()` and `onAuthenticationFailure()`: these are 2 jobs that all authentication should think about (even if success simply does nothing - and even in that case, maybe you use it as a convenient spot to do some logging or call some other custom code in your app to do something - that could be done in a listener, but it's also valid to do it here).